sycl/test/warnings/sycl_2020_deprecations.cpp

```diff
@@ -6,6 +6,8 @@
 #include <CL/sycl.hpp>
 #include <sycl/ext/intel/online_compiler.hpp>
 
+class Test;
```

Comment:
Should this test fail w/o the implementation from this https://github.com/intel/llvm/pull/4945 patch?

Reply:
I guess no.

Reply:
I didn't realize we could even run host-only mode without integration header. Are we breaking valid code then?

Reply:
I'm not sure if we breaking it. This test tests compilation diagnostics, so it doesn't run anything.
In my understanding any code in sycl host only mode that runs kernels is not viable without integration header.

Reply:
Yea I would expect that integration header is mandatory, and not providing one is undefined behavior or error.  @bader can you confirm?

Reply:
@Fznamznon, follow up question: if the test will not fail w/o the patch and it still passes with this patch, mb this line should be removed?

Reply:
I think someone was confused (maybe me). The test won't fail w/o the patch, but It fails if not changed with this patch. If I remove this line, It will start failing because of additional diagnostic.

Reply:
Oh, thanks, it's clear now for me.

Reply:
> Yea I would expect that integration header is mandatory, and not providing one is undefined behavior or error. @bader can you confirm?

According to my understanding, integration header is required only for the SYCL mode when source code is compiled for additional target. It should not be required if we compile an application in C++ mode.

Reply:
Do we consider the situation when flag `fsycl-is-host` is passed as compilation in C++ mode?

Reply:
According to my understanding `fsycl-is-host` enables SYCL language mode, the same way `fsycl-is-device` does.

Reply:
Then, I guess we should be ok. The new error is emitted only if `fsycl-is-host` is passed.